crates/fs/benches/namefilter.rs

```diff
@@ -0,0 +1,72 @@
+use criterion::{black_box, criterion_group, criterion_main, Criterion};
```

Comment:
we should prob measure and compare in CI. One example action out there is https://github.com/boa-dev/criterion-compare-action, but we can build it ourselves using critcmp: https://github.com/BurntSushi/critcmp. Actually, many use https://github.com/benchmark-action/github-action-benchmark, so maybe that's the way to go these days. 

As an example of the github-action-benchmark and also uploading of criterion artifacts (which are nice): https://github.com/osohq/oso/blob/main/.github/workflows/bench.yml#L29.